src/query/api/v1/handler/prom/common.go

```diff
@@ -92,16 +92,5 @@ func Respond(w http.ResponseWriter, data interface{}, warnings promstorage.Warni
 
 // Responds with error status code and writes error JSON to response body.
 func RespondError(w http.ResponseWriter, err error) {
-	json := jsoniter.ConfigCompatibleWithStandardLibrary
-	b, marshalErr := json.Marshal(&response{
-		Status: statusError,
-		Error:  err.Error(),
-	})
-	if marshalErr != nil {
-		xhttp.WriteError(w, marshalErr)
-		return
-	}
-
-	w.Header().Set(xhttp.HeaderContentType, xhttp.ContentTypeJSON)
-	xhttp.WriteError(w, err, xhttp.WithErrorResponse(b))
+	xhttp.WriteError(w, err)
```

Comment:
Now that this just calls WriteError can we remove it and update the callers in place to use `xhttp.WriteError` directly?